modules/mkdru_ding/js/mkdru_ding.theme.js

```diff
@@ -0,0 +1,256 @@
+    document.getElementById("full_" + id).style.display = 'block';
+  };
+  Drupal.theme.mkdruShowShortDescr = function (id) {
+    Drupal.TingSearchOverlay(false);
+    document.getElementById("short_" + id).style.display = 'block';
+    document.getElementById("full_" + id).style.display = 'none';
+  };
+  Drupal.theme.mkdruTruncateDescr = function (desc, length) {
+    var s = desc.substr(0, length);
+    return s.substr(0, s.lastIndexOf(' '));
+  };
+  Drupal.theme.mkdruSafeTrim = function (s) {
+    if (s.trim) {
+      return s.trim();
+    }
+    if (s.replace) {
+      return s.replace(/^\s+|\s+$/g, "");
+    }
+    // argh
+    if (console && console.log) {
+      console.log("String object doesn't even have replace() ??");
+    }
+    return s;
+  };
+
+  Drupal.theme.mkdruResult = function (hit, num, detailLink) {
+    var link = choose_url(hit);
+    var basePath = Drupal.settings.basePath;
+    var specific_author_field = "";
+    var specific_subject_field = "";
+    if (mkdru.settings) {
+      if (mkdru.settings.specific_author_field) {
+        specific_author_field = mkdru.settings.specific_author_field + '=';
+      }
+      if (mkdru.settings.specific_subject_field) {
+        specific_subject_field = mkdru.settings.specific_subject_field + '=';
+      }
+    }
+
+    if (!link) {
+      link = choose_url(hit.location[0]);
+    }
+    var html = "";
+
+    html += '<li class="search-result" id="rec_' + hit.recid + '" ><div class="result-inner-wrapper">' +
+      '<h3 class="title">';
+    if (link) {
+      html += '<a href="' + link + '" target="_blank" >';
+    }
+    html += hit["md-title"];
+    if (link) {
+      html += '</a>';
+    }
+    if (hit.location[0]['md-medium']) {
+      html += " (" + hit.location[0]['md-medium'] + ")";
+    }
+    html += '</h3>';
+
+    // Add target to each result item.
+    var target = '';
+    if (typeof (hit.location[0]['@name']) !== 'undefined') {
+      target =  hit.location[0]['@name'];
+    }
+
+    html += '<div class="mkdru-target">' + target + '</div>';
+    html += '<div class="search-snippet-info">' +
+      '<p class="search-snippet"></p>' +
+      '<div class="ting-object clearfix">' +
+      '<div class="ting-overview clearfix">' +
+      '<div class="left-column left">' +
+      '<div class="picture">';
+
+    if (hit['md-thumburl']) {
+      html += '<img src="' + hit['md-thumburl'][0] + '" alt="' + hit['md-title'] + '" />';
+    }
+
+    html += '</div>' +
+      '</div>' +
+      '<div class="right-column left">';
+
+    if (hit["md-author"]) {
+      // expand on ; and reprint in the same form
+      var authors = hit["md-author"][0].split(';');
+      html += '<div class="creator"><span class="byline">' + Drupal.t('By') + ' </span>';
+      for (var i = 0; i < authors.length - 1; i++) {
+        html += '<a class="author" href="' + basePath + 'search/meta/' + specific_author_field + Drupal.theme.mkdruSafeTrim(authors[i]) + '">' + authors[i] + '</a> ;';
```

Comment:
href needs Drupal.settings.basePath + Drupal.settings.pathPrefix to work consistently